sklearn/utils/estimator_checks.py

```diff
@@ -1732,6 +1732,25 @@ def check_transformer_preserve_dtypes(name, transformer_orig):
         )
 
 
+def check_regressor_preserve_dtypes(name, regressor_orig):
+    # check that dtype are preserved meaning if label y is of some dtype
+    # y_pred should be from the same dtype.
+    X, y = make_regression(random_state=42, n_targets=5, n_samples=100, n_features=3)
+    X = StandardScaler().fit_transform(X)
+
+    for dtype in _safe_tags(regressor_orig, key="preserves_dtype"):
+        y_cast = y.astype(dtype)
+        regressor = clone(regressor_orig)
+        regressor.fit(X, y_cast)
+        y_pred = regressor.predict(X)
+
+        # check that the output dtype is preserved
+        assert y_pred.dtype == dtype, (
+            f"Regressor predicted dtype: {y_pred.dtype} - "
+            f"original/expected dtype: {dtype.__name__}"
+        )
```

Comment:
```suggestion
        assert y_pred.dtype == dtype, (
            f"{regressor.__class__.__name__}.predict returns"
            f"{y_pred.dtype.__name__} predictions instead of" 
            f"{dtype.__name__} predictions."
        )
```